envoy/router/router.h

```diff
@@ -1547,7 +1547,7 @@ class GenericUpstream {
    * will not fully close the connection. This is off by default.
    * @param enabled Whether to set half-close semantics as enabled or disabled.
    */
-  virtual void enableHalfClose() PURE;
```

Comment:
Mind splitting this out?  Adding renames to a complicated PR is just going to make it harder to land